lib/sedna/backend/mindspore/__init__.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 The KubeEdge Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import mindspore.context as context
+from sedna.backend.base import BackendBase
+from sedna.common.file_ops import FileOps
+
+
+class MSBackend(BackendBase):
+    def __init__(self, estimator, fine_tune=True, **kwargs):
+        super(MSBackend, self).__init__(estimator=estimator,
+                                        fine_tune=fine_tune,
+                                        **kwargs)
+        self.framework = "mindspore"
+
+        if self.use_npu:
+            context.set_context(mode=context.GRAPH_MODE, device_target="Ascend")
+        elif self.use_cuda:
+            context.set_context(mode=context.GRAPH_MODE, device_target="GPU")
+        else:
+            context.set_context(mode=context.GRAPH_MODE, device_target="CPU")
+
+        if callable(self.estimator):
+            self.estimator = self.estimator()
+
+    def train(self, train_data, valid_data=None, **kwargs):
```

Comment:
I tried your method, but it didn't work very well. That is because, for class `mindspore.Model`, the data passed to it MUST BE `mindspore.Dataset'[https://www.mindspore.cn/docs/api/zh-CN/r1.3/api_python/mindspore.html?highlight=train_dataset#mindspore.Model.train](url), and can not be a BaseDataSource.
Here are some solutions I thought of：
1. Abandon class `mindspore.Model` and adopt a more flexible approach: `for data in dataloader: loss = network(data), loss.update()`.
2. Repackage `train_data` into a `mindspore.Dataset` in `interface.py`.
For solution 1, it increases the difficulty of model development and is not a popular way.
For solution 2, actually it is counterintuitive.
Which do you prefer? Or do you have a more reasonable solution?

Reply:
I tried your method, but it didn't work very well. That is because, for method `mindspore.Model.train`, the data passed to it MUST BE a `mindspore.Dataset', and can not be a `BaseDataSource`.
Here are some solutions I thought of：
1. Abandon class `mindspore.Model` and adopt a more flexible approach: `for data in dataloader: loss = network(data), loss.update()`.
2. Repackage `train_data` into a `mindspore.Dataset` in `interface.py`.
For solution 1, it increases the difficulty of model development and is not a popular way.
For solution 2, actually it is counterintuitive.
Which do you prefer? Or do you have a more reasonable solution?

Reply:
I tried your method, but it didn't work very well. That is because, for class `mindspore.Model`, the data passed to it MUST BE `mindspore.Dataset'[https://www.mindspore.cn/docs/api/zh-CN/r1.3/api_python/mindspore.html?highlight=train_dataset#mindspore.Model.train](url), and can not be a BaseDataSource.
Here are some solutions I thought of：

- 1. Abandon class `mindspore.Model` and adopt a more flexible approach: `for data in dataloader: loss = network(data), loss.update()`.
- 2. Repackage `train_data` into a `mindspore.Dataset` in `interface.py`.

For solution 1, it increases the difficulty of model development and is not a popular way.
For solution 2, actually it is counterintuitive.
Which do you prefer? Or do you have a more reasonable solution?

Reply:
I tried your method, but it didn't work very well. That is because, for method `mindspore.Model.train`, the data passed to it MUST BE a `mindspore.Dataset', and can not be a `BaseDataSource`.
Here are some solutions I thought of：

- 1. Abandon class `mindspore.Model` and adopt a more flexible approach: `for data in dataloader: loss = network(data), loss.update()`.
- 2. Repackage `train_data` into a `mindspore.Dataset` in `interface.py`.

For solution 1, it increases the difficulty of model development and is not a popular way.
For solution 2, actually it is counterintuitive.
Which do you prefer? Or do you have a more reasonable solution?

Reply:
I tried your method, but it didn't work very well. That is because, for method `mindspore.Model.train`, the data passed to it MUST BE a `mindspore.Dataset`, and can not be a `BaseDataSource`.
Here are some solutions I thought of：

- 1. Abandon class `mindspore.Model` and adopt a more flexible approach: `for data in dataloader: loss = network(data), loss.update()`.
- 2. Repackage `train_data` into a `mindspore.Dataset` in `interface.py`.

For solution 1, it increases the difficulty of model development and is not a popular way.
For solution 2, actually it is counterintuitive.
Which do you prefer? Or do you have a more reasonable solution?

Reply:
PTAL @jaypume

Reply:
I suggest that `BaseDataSorce` is designed to compatible with existing `Dataset` such as `mindspore.Dataset`, `torch.utils.data.Dataset`, `tf.data.Dataset`. If the Dataset instance is passed to for example `FederatedLearning.train(tfDatasetInstance)`, it should work.

Reply:
> I suggest that `BaseDataSorce` is designed to compatible with existing `Dataset` such as `mindspore.Dataset`, `torch.utils.data.Dataset`, `tf.data.Dataset`. If the Dataset instance is passed to for example `FederatedLearning.train(tfDatasetInstance)`, it should work.

You can help us raise an issue so we can track it.